lisp/ess-inf.el

```diff
@@ -241,7 +231,8 @@ This may be useful for debugging."
             (ess-wait-for-process proc)))))))
 
 (defun inferior-ess--get-proc-buffer-create (proc-name)
-  "Get a process buffer, creating a new one if needed."
+  "Get a process buffer, creating a new one if needed.
+This always returns a process-less buffer."
```

Comment:
I don't think that this function does that. It just returns the buffer `inf-name`. If `inf-name` buffer has process, it doesn't check it.

Reply:
yes but `inf-name` will never have a live process in it. I wanted to pull the loop that checks for this into this function, but `proc-name` is needed in the caller. I can add a redundant check just for readability.

Reply:
hmm or is it possible that `ess-gen-proc-buffer-name-function` may return a buffer name with a live process?

But then is there something wrong with that loop?

```lisp
           (proc-name (let ((ntry 1))
                        (while (get-process (ess-proc-name ntry temp-dialect))
                          (setq ntry (1+ ntry)))
                        (ess-proc-name ntry temp-dialect)))
```

Should it call the name generating function before checking for process?

Reply:
eturn a buffer name with a live process 

Potentially yes. But, that loop is for process name, which always stays the same. The gen function is for buf names. As our default buf-generators include proc-name (right?) the buf names are probably unique unless someone changed them by hand.

I cannot help more as I write from memory of that code. Just do whatever feels right but please make that new function self-contained somehow, even if that requires throwing.